pkg/core/manager/extension.go

```diff
@@ -0,0 +1,230 @@
+				ExtensionID: extensionID,
+				ServiceID:   service.ServiceID,
+			})
+		if err != nil {
+			return nil, err
+		}
+	}
+	return &extRecord, nil
+}
+
+// GetService - retrieve an extension service by ID and, optionally, its entire endpoint/credentials subtree
+func (registry *ExtensionRegistry) GetService(ctx context.Context, serviceID domain.ExtensionServiceID, fullTree bool) (result *domain.ExtensionServiceRecord, err error) {
+	service, err := registry.extensionStore.GetService(ctx, serviceID)
+	if err != nil {
+		return nil, err
+	}
+	svcRecord := domain.ExtensionServiceRecord{
+		ExtensionService: *service,
+		Endpoints:        make([]*domain.ExtensionEndpoint, 0),
+		Credentials:      make([]*domain.ExtensionCredentials, 0),
+	}
+	if !fullTree {
+		return &svcRecord, nil
+	}
+	svcRecord.Endpoints, err = registry.extensionStore.GetServiceEndpoints(ctx, serviceID)
+	if err != nil {
+		return nil, err
+	}
+
+	svcRecord.Credentials, err = registry.extensionStore.GetServiceCredentials(ctx, serviceID)
+	if err != nil {
+		return nil, err
+	}
+	return &svcRecord, nil
+}
+
+// GetEndpoint - retrieve an extension endpoint by ID
+func (registry *ExtensionRegistry) GetEndpoint(ctx context.Context, endpointID domain.ExtensionEndpointID) (result *domain.ExtensionEndpoint, err error) {
+	return registry.extensionStore.GetEndpoint(ctx, endpointID)
+}
+
+// GetCredentials - retrieve a set of extension credentials by ID
+func (registry *ExtensionRegistry) GetCredentials(ctx context.Context, credentialsID domain.ExtensionCredentialsID) (result *domain.ExtensionCredentials, err error) {
+	return registry.extensionStore.GetCredentials(ctx, credentialsID)
+}
+
+// RemoveExtension - remove an extension from the registry
+func (registry *ExtensionRegistry) RemoveExtension(ctx context.Context, extensionID string) error {
+	return registry.extensionStore.DeleteExtension(ctx, extensionID)
+}
+
+// RemoveService - remove an extension service from the registry
+func (registry *ExtensionRegistry) RemoveService(ctx context.Context, serviceID domain.ExtensionServiceID) error {
+	return registry.extensionStore.DeleteService(ctx, serviceID)
+}
+
+// RemoveEndpoint - remove an extension endpoint from the registry
+func (registry *ExtensionRegistry) RemoveEndpoint(ctx context.Context, endpointID domain.ExtensionEndpointID) error {
+	return registry.extensionStore.DeleteEndpoint(ctx, endpointID)
+}
+
+// RemoveCredentials - remove a set of extension credentials from the registry
+func (registry *ExtensionRegistry) RemoveCredentials(ctx context.Context, credentialsID domain.ExtensionCredentialsID) error {
+	return registry.extensionStore.DeleteCredentials(ctx, credentialsID)
+}
+
+// RunQuery - run a query on the extension registry to find one or more ways to access extensions matching given search parameters
+func (registry *ExtensionRegistry) RunQuery(ctx context.Context, query *domain.ExtensionQuery) (result []*domain.ExtensionAccessDescriptor, err error) {
```

Comment:
I think `RunQuery` is a very generic name, if the function returns a list of `ExtensionAccessDescriptor`, the function should at least mean that (e.g. `GetExtensionAccess`.

Reply:
Done. Renamed to `RunExtensionAccessQuery`